plugins/in_disk/in_disk.c

```diff
@@ -342,6 +331,26 @@ static int in_disk_exit(void *data, struct flb_config *config)
     return 0;
 }
 
+/* Configuration properties map */
+static struct flb_config_map config_map[] = {
+    {
+      FLB_CONFIG_MAP_INT, "interval_sec", DEFAULT_INTERVAL_SEC,
+      0, FLB_TRUE, offsetof(struct flb_in_disk_config, interval_sec),
+      "Set the collector interval"
+    },
+    {
+      FLB_CONFIG_MAP_INT, "interval_nsec", DEFAULT_INTERVAL_NSEC,
+      0, FLB_TRUE, offsetof(struct flb_in_disk_config, interval_nsec),
+      "Set the collector interval (sub seconds)"
+    },
+    {
+      FLB_CONFIG_MAP_STR, "dev_name", "",
+      0, FLB_TRUE, offsetof(struct flb_in_disk_config, dev_name),
```

Comment:
I think the data type of `dev_name` should be `flb_sds_t`.

Reply:
Does flb_input_config_map_set use flb_sds_t when handling strings?

Reply:
I think so.

https://github.com/fluent/fluent-bit/blob/v1.8.12/src/flb_config_map.c#L353
This is a part of destructor of config_map.
It calls `flb_sds_destroy` for `FLB_CONFIG_MAP_STR`.

Reply:
https://github.com/fluent/fluent-bit/blob/v1.8.12/include/fluent-bit/flb_config_map.h#L58
It is the define of `struct flb_config_map_val`.
It uses `flb_sds_t`.

Reply:
I think it's handled in the background by flb_config_map_set() [flb_config_map.c#L613](https://github.com/fluent/fluent-bit/blob/9c2868fc841e1f0d9e0fd96a4bd8da164741906c/src/flb_config_map.c#L613):

```c
        /* All the following steps are direct writes to the user context */
        if (m->type == FLB_CONFIG_MAP_STR) {
            m_str = (char **) (base + m->offset);
            *m_str = m->value.val.str;
        }
```

So the struct's property type should be `char *` but it is in fact being handled as a flb_sds_t internally, as far as I can tell.